### PR TITLE
Maintenancy realignment

### PR DESCRIPTION
**Pull request by ggarri**
## PR description
- Aligned migrations with previous versions
- Cherry-picked cache manager component
- Cherry-picked storage of contractName and contractTag as part of transaction data for data optimization proposes
- Aligned circleCI config from maintenance

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://github.com/consensys/orchestrate/blob/master/CHANGELOG.md).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -234,6 +237,7 @@ jobs:
           docker-user: << parameters.docker-user >>
           docker-image-tag: << parameters.docker-image-tag >>
 
+
   run-e2e:
```
Comment (darioAnongba) [inline .circleci/config.yml:241]: run-e2e is different on main, please keep the old job as you don't need cucumber and have no report

File: .circleci/config.yml [hunk 2]
```diff
@@ -253,17 +257,25 @@ jobs:
       - run:
           name: Running e2e
           environment:
+            CUCUMBER_OUTPUTPATH: "/report/output/report.json"
```
Comment (darioAnongba) [inline .circleci/config.yml:260]: no cucumber needed

File: .circleci/config.yml [hunk 3]
```diff
@@ -253,17 +257,25 @@ jobs:
...
+            CUCUMBER_FORMAT: cucumber
+            CUCUMBER_STEPS_TIMEOUT: 30s
             KAFKA_CONSUMER_GROUP_NAME: e2e
             KAFKA_TLS_ENABLED: true
             KAFKA_TLS_INSECURE_SKIP_VERIFY: true
             KAFKA_SASL_ENABLED: true
             KAFKA_SASL_MECHANISM: PLAIN
-          command: make run-e2e
+          command: make e2e-ci
+      - store_artifacts:
```
Comment (darioAnongba) [inline .circleci/config.yml:271]: no artifacts

File: Makefile [hunk 4]
```diff
@@ -17,11 +17,6 @@ ifeq ($(UNAME_S),Darwin)
 	VEGETA_BIN_URL = https://github.com/tsenart/vegeta/releases/download/v12.8.4/vegeta_12.8.4_darwin_amd64.tar.gz
 endif
 
-ifneq (,$(wildcard ./.env))
```
Comment (darioAnongba) [inline Makefile:20]: why removed?

File: docker-compose.yml [hunk 5]
```diff
@@ -160,9 +181,10 @@ services:
         EOF
 
   key-manager-migration:
-    image: consensys/quorum-key-manager:v21.12.2
+    image: consensys/quorum-key-manager:v21.12.0
```
Comment (darioAnongba) [inline docker-compose.yml:184]: why not latest QKM

File: docker-compose.yml [hunk 6]
```diff
@@ -171,9 +193,10 @@ services:
       - postgres
 
   key-manager:
-    image: consensys/quorum-key-manager:v21.12.2
+    image: consensys/quorum-key-manager:v21.12.0
```
Comment (darioAnongba) [inline docker-compose.yml:196]: why not latest QKM

File: src/api/store/models/transaction.go [hunk 7]
```diff
@@ -81,6 +85,8 @@ func (tx *Transaction) ToEntity() *entities.ETHTransaction {
 		PrivateFor:      tx.PrivateFor,
 		MandatoryFor:    tx.MandatoryFor,
 		PrivacyGroupID:  tx.PrivacyGroupID,
+		ContractName:    tx.ContractName,
```
Comment (darioAnongba) [inline src/api/store/models/transaction.go:88]: Don't see the need for this in main as we are going to refactor the contract registry and the schedules so this info is part of the tx request already.

File: src/entities/eth_transaction.go [hunk 8]
```diff
@@ -43,6 +43,8 @@ type ETHTransaction struct {
 	GasTipCap       *hexutil.Big
 	AccessList      types.AccessList
 	TransactionType TransactionType
+	ContractName    string
```
Comment (darioAnongba) [inline src/entities/eth_transaction.go:46]: same remark here

File: src/tx-listener/tx-listener/sessions/chains/session_test.go [hunk 9]
```diff
@@ -1,5 +1,4 @@
 // +build unit
-// +build !race
```
Comment (darioAnongba) [inline src/tx-listener/tx-listener/sessions/chains/session_test.go:2]: I thought you asked me to add this last time

File: .circleci/config.yml (hunk 1 above)
Comment (ggarri) [reply .circleci/config.yml:241]: indeed, changes were cherry-pick and I forgot to review this

File: .circleci/config.yml (hunk 2 above)
Comment (ggarri) [reply .circleci/config.yml:260]: removed

File: Makefile (hunk 4 above)
Comment (ggarri) [reply Makefile:20]: applying what you have in your `.env` to your global env it leads to unexpected behaviours like having variables set when you run other commands like `make run-unit` which make tests to fail

File: docker-compose.yml (hunk 5 above)
Comment (ggarri) [reply docker-compose.yml:184]: back ported by accident

File: src/entities/eth_transaction.go (hunk 8 above)
Comment (ggarri) [reply src/entities/eth_transaction.go:46]: same about keeping track of work and migrations

File: src/tx-listener/tx-listener/sessions/chains/session_test.go (hunk 9 above)
Comment (ggarri) [reply src/tx-listener/tx-listener/sessions/chains/session_test.go]: reverted

**Review by darioAnongba — APPROVED** [review]

**Pull request merged**